Lazy_Lists.md

```diff
@@ -0,0 +1,41 @@
+# Lazy Lists
+
+Lazy lists are one of the fundamental aspects in Functional Programming because it allows us to define and create infinite lists.  In Coconut, Lazy lists, represented by “banana brackets” `(| |)`, are defined to be lazily evaluated iterator literals. This means that values are not stored in the list, but iterables are. Lazy evaluation is similar to the `::` operator, in that expressions are not evaluated until that particular element is needed. 
```

Comment:
Since we mentioned the banana brackets here and the :: operator later, would it make sense to mention the :: operator here as well?

Reply:
I wasn't sure if `::` was mentioned in another section, but it would make sense to do a super brief description of the `::` operator here.